raiden/tests/integration/api/test_restapi.py

```diff
@@ -1677,13 +1678,57 @@ def test_register_token(
     conflict_response = conflict_request.send().response
     assert_response_with_error(conflict_response, HTTPStatus.CONFLICT)
 
+    # Test that adding a second token throws a forbidden error
+    forbidden_request = grequests.put(
+        api_url_for(
+            api_server_test_instance,
+            "registertokenresource",
+            token_address=to_checksum_address(other_token_address),
+        )
+    )
+    forbidden_response = forbidden_request.send().response
+    assert_response_with_error(forbidden_response, HTTPStatus.FORBIDDEN)
+    response_json = get_json_response(forbidden_response)
+    assert "Number of token networks will exceed the maximum of" in response_json["errors"]
+
+
+@raise_on_failure
+@pytest.mark.parametrize("number_of_tokens", [0])
+@pytest.mark.parametrize("number_of_nodes", [1])
+@pytest.mark.parametrize("channels_per_node", [0])
+@pytest.mark.parametrize("environment_type", [Environment.DEVELOPMENT])
+def test_register_token_without_balance(
```

Comment:
Currently, our fixtures take quite long to be set up. So splitting a test increases the run time. I'm not sure about the best trade-off for this.

Reply:
This is hard to put into one test, as now the  `MaxTokenNetworkNumberReached` is triggered before. Lets optimize this another day.